collect_app/src/main/java/org/odk/collect/android/geo/MapsMarkerCache.kt

```diff
@@ -0,0 +1,31 @@
+package org.odk.collect.android.geo
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.util.LruCache
+import androidx.annotation.DrawableRes
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.toBitmap
+
+object MapsMarkerCache {
+    /**
+     * We use different markers in different features but it looks like we will never need to
+     * support more than 10 different types of markers at the same time.
+     */
+    private val cache = LruCache<Int, Bitmap>(10)
```

Comment:
I guess we're thinking that we should always have enough memory for all 10 right? If we don't, I guess we won't be able to render the map (with all 10 types).

Reply:
Yes 10 doesn't seem like a big number especially taking into account that our markers are not huge. In many cases we are not going to need all 10. For example in case of:
- Geotrace widget we use just one marker which is a red circle
- Geoshape widgets the same as above
- FormMapActivity it depends in how many states your form are saved we have 4 states and every state might have normal and big marker (the big one is displayed if you click on it) so it's 8 + 1 (for the default red circle marker which is displayed by default and then replaced by one of the states - this is something we can improve in the future I think) = 9

I don't know what would happen if we had such a poor device that can't handle up to 9 bitmaps, probably the same exception would be thrown.